### ISSUE TITLE
Support mods loaded from the support directory

### ISSUE THREAD

**Issue by pchote** [OpenRA/OpenRA#8992]
This is another fundamental engine feature for third party mods.  Expecting players to install mods in the main game dir is dangerous, and causes trouble when we release new engine versions.  Loading mods from the support dir fixes most of these problems.

We should support both unpacked mods (in a directory) and packaged mods in a zip renamed to oramod.
Because these mods are effectively data files in a user-owned directory there are no fundamental problems with supporting in-game installation and upgrades for them.

The tricky part is dealing with file paths, as our current mod.yaml syntax defines only `.` for "relative to game exe" and `^` for "relative to support dir".  I propose introducing a `<mod id>:` prefix for referencing files relative to the current or another named mod dir.

e.g. a snippet of the d2k mod may look like:
```yaml
Assemblies:
	common: OpenRA.Mods.Common.dll
	ra: OpenRA.Mods.RA.dll
	cnc: OpenRA.Mods.Cnc.dll
	d2k: OpenRA.Mods.D2k.dll

ChromeLayout:
	d2k: chrome/install.yaml
	d2k: chrome/ingame.yaml
	ra: chrome/ingame-chat.yaml
	ra: chrome/ingame-diplomacy.yaml
	ra: chrome/ingame-fmvplayer.yaml
	d2k: chrome/ingame-menu.yaml
	ra: chrome/ingame-info.yaml
	ra: chrome/ingame-infoscripterror.yaml
```

**Comment by penev92** [comment]
As a long-term solution I'd like for the mod chooser to be able to install mods (not assets).
This will include browsing mods hosted on http://www.resource.openra.net/ and downloading a selected mod into the `OpenRA\mods\` directory.
That would also depend on #8991, of course.

This could work as a short-term solution though.

**Comment by pchote** [comment]
Installing mods into the main installation directory is probably never going to happen.  Operating system permissions prevent us from writing or modifying files there, and even if we did implement permission escalation for all our supported platforms it violates the platform standards under at least Linux.

**Comment by penev92** [comment]
OK, I'd still like to see that happen though, even if the mods get installed into the user directory.

**Comment by GraionDilach** [comment]
@penev92 Bad idea, third-party mods can be really large at filesize. The bigger TS/YR mods nowadays (Mental Omega, Twisted Insurrection, Robot Storm, Project Phantom, D-Day, etc) are atleast 200-300 MB, which has the danger of running the resource server out of it's quotas.

**Comment by penev92** [comment]
It can always link to downloads hosted on ModDB/wherever. The point is we lack a centralized place to browse mods from, as have all C&C games, and ModDB doesn't quite cut it IMO.

**Comment by Unit158** [comment]
I think if we use the resource site as a browser and require mod devs (perhaps other large assets as well?) to find their own hosting. Also, what prevents a user from uploading a 3 GB map atm?

**Comment by GraionDilach** [comment]
#8999. While not the best prevention method, it technically does that.

**Comment by Phrohdoh** [comment]
That does not at all prevent a user from uploading a file of N byte size.
@Unit158 if we want to limit upload size we can do it, that isn't anything new to the web.

**Comment by Mailaender** [comment]
Duplicate of https://github.com/OpenRA/OpenRA/issues/2426.

**Comment by Phrohdoh** [comment]
This ticket is more complete and thorough, reverse closing because of this.

**Comment by atlimit8** [comment]
`:` suffix looks like a key-value pair.
Perhaps something like `$`*mod-id*`/`*path* would be more distinguishable.
The (UNIX) shell environment variable feel is intentional.
